### DISCUSSION TITLE
Getting 'Error writing to file' during WiX Toolset upgrade when changing installation scope

### DISCUSSION THREAD

**Discussion by dotriz** [wixtoolset/issues#8087] [Questions]
I recently upgraded my application's installer created with WiX Toolset 4. In the older version, I did not specify a scope, while in the newer version, I set the scope to "perUser" to allow Non-Admin users to install the app aswell.

However, when attempting to update the application from the old version to the new one using the new installer, I encounter errors like

> "Error writing to file filename.exe. Verify that you have access to the directory."

Both the old and new setups are installed in the same folder within **AppData**. Additionally, I use the `<MajorUpgrade>` element in my setup to ensure that the previous version is uninstalled before installing the new one. Despite this, I'm still facing this issue.

Any insights or suggestions on how to troubleshoot and resolve this problem would be greatly appreciated. Thank you!
[installation-log.txt](https://github.com/wixtoolset/issues/files/14828641/installation-log.txt)

**Comment by wmanning** [comment]
Did you try running the installer as an admin as a test?  

I think the default scope is perMachine which requires admin access.  So the new perUser installer would require admin access to uninstall the old per-Machine MSI before installing the new per-User MSI.

**Reply by dotriz** [reply]
Yes, I did run it as an administrator, and it installs fine. However, the reason for making it a per-user installer is to allow non-admin users to install it as well. How can we achieve this to update existing users smoothly?

**Reply by wmanning** [reply]
> How can we achieve this to update existing users smoothly?

I don't think there is a way. They'll have to be an admin to uninstall the existing MSI.  Going forward at least they won't have to be but will be that initial pain point.
